src/main/java/com/palantir/gradle/externalpublish/ExternalPublishIntellijPlugin.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.externalpublish;
+
+import java.lang.reflect.InvocationTargetException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.JavaExec;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.jvm.toolchain.JavaLauncher;
+import org.jetbrains.intellij.tasks.BuildPluginTask;
+import org.jetbrains.intellij.tasks.PatchPluginXmlTask;
+import org.jetbrains.intellij.tasks.PublishPluginTask;
+
+public class ExternalPublishIntellijPlugin implements Plugin<Project> {
+
+    @Override
+    public final void apply(Project project) {
+
+        project.getPlugins().apply("org.jetbrains.intellij");
+
+        TaskProvider<PublishPluginTask> publishPlugin =
+                project.getTasks().named("publishPlugin", PublishPluginTask.class);
+
+        TaskProvider<BuildPluginTask> buildPlugin = project.getTasks().named("buildPlugin", BuildPluginTask.class);
+
+        TaskProvider<PatchPluginXmlTask> patchPluginXml =
+                project.getTasks().named("patchPluginXml", PatchPluginXmlTask.class);
+
+        patchPluginXml.configure(task -> {
+            task.getVersion().set(project.getVersion().toString());
+        });
+
+        project.getTasks().withType(JavaExec.class).named("runIde", task -> {
+            task.jvmArgs("--add-exports=java.base/jdk.internal.vm=ALL-UNNAMED");
+            task.dependsOn(buildPlugin);
+        });
+
+        project.afterEvaluate(projectAfterEvaluate -> {
+            projectAfterEvaluate.getTasks().withType(JavaExec.class).named("runIde", task -> {
+                task.getJavaLauncher().set((JavaLauncher) null);
+            });
+        });
+
+        publishPlugin.configure(task -> {
+            task.onlyIf(_ignored -> OurEnvironmentVariables.isTagBuild(project));
+            task.getToken().set(System.getenv("JETBRAINS_PLUGIN_REPO_TOKEN"));
+        });
+
+        project.afterEvaluate(_ignored -> {
+            project.getTasks().named("publish", task -> {
+                task.dependsOn(publishPlugin);
+            });
+        });
+
+        project.getTasks().named("check", task -> {
```

Comment:
It's best practice to apply the `LifecycleBasePlugin.class` before using [one of these tasks](https://github.com/gradle/gradle/blob/72b4352a86b96267249412535aeedf63aab9d8bd/platforms/software/platform-base/src/main/java/org/gradle/language/base/plugins/LifecycleBasePlugin.java#L34-L37) it makes.